client/src/components/BookList/BookListShowPage.js

```diff
@@ -1,47 +1,67 @@
 import React, { useState, useEffect } from "react";
 import { withRouter } from "react-router-dom";
 import BookTile from "../Book/BookTile.js";
-
+import BookListIndex from "../BookList/BookListIndex.js";
 const showBooksBookList = (props) => {
   const [books, setBooks] = useState([]);
   const [bookListName, setBookListName] = useState("");
-  const [bookFavorites, setBookFavorites] = useState("");
+  const [bookLists, setBookLists] = useState([]);
   const bookListId = props.match.params.id;
 
-  const getBooks = async () => {
+  const getBookListName = async () => {
     try {
-      const response = await fetch(`/api/v1/bookLists/${bookListId}`);
+      const response = await fetch(`/api/v1/bookLists`);
       debugger;
       if (!response.ok) {
         const errorMessage = `${response.status} (${response.statusText})`;
         const error = new Error(errorMessage);
         throw error;
       }
 
+      const bookListData = await response.json();
+      bookListData.bookLists.forEach((bookLists) => {
+        if (bookListId === bookLists.id) {
+          return setBookListName(bookLists.name);
+        }
+      });
```

Comment:
this works, but we shouldnt be getting ALL of the booklists from the backend here, we should only be getting a specific book list. So your original `await fetch(`/api/v1/bookLists/${bookListId}`);` is the way you want to go

Reply:
you then want to use a serialize to get all of the books in this book list, so that you dont have to have `getBooks`. Ill comment in your router, but you will want to change these things, otherwise this will eventually break